tianshou/policy/base.py

```diff
@@ -150,7 +185,21 @@ def forward(
         """
         pass
 
-    def map_action(self, act: Union[Batch, np.ndarray]) -> Union[Batch, np.ndarray]:
+    @overload
```

Comment:
Since the PR is already big, I would postpone it. There are many methods in tianshou that deserve an overload, especially ones that act on unions of arrays and tensors. I would do that in a future PR, if you don't mind

Reply:
sgtm, thanks in advance and please create an issue!